Assets/Scripts/Game/Phase/Timeline/Timeline.cs

```diff
@@ -36,18 +34,26 @@ public class Timeline : Phase {
 			foreach(Unit unit in player.units) {
 				if(unit.plan.Count > 0) {
 					ActionType actionType = unit.plan.Peek().type;
-                    Debug.Log(actionType);
                     actionDict[actionType].Enqueue(unit);
 				}
 			}
 		}
 
 		//Play Execution Based on Order
 		foreach(ActionType type in actionOrder) {
-			Queue<Unit> actions = actionDict[type];
-			while(actions.Count > 0) {
-				Unit unit = actions.Dequeue();
-				ExecuteUnitCommand(unit, board);
+			Queue<Unit> actions = FilterCanExecute(actionDict[type], board);
+			HashSet<Unit> toExecute = UnitsToExecute(actions);
+			Dictionary<Unit, UnitDisplacement> displacements = GetDisplacements(players, actions, board);
+			List<SimulatedDisplacement> result = SimulateDisplacements(displacements, board);
+
+			foreach(SimulatedDisplacement sim in result) {
+				Unit unit = sim.displacement.unit;
+				if(toExecute.Contains(unit)) {
+					Command command = unit.plan.Dequeue();
```

Comment:
Units that are filtered out by FilterCanExecute don't have their next command dequeued, and thus get stuck trying to perform their next command until it becomes valid. e.g. if a unit is at 0,0 and their plan is move up once (an invalid move) followed by valid moves (moving right, attacking, etc.) they won't do anything. Not sure if this is intended behaviour.

Reply:
No this is not intended behaviour

Reply:
I've decided to move the filterExecuteAction check to after the simulation. The original intent of the filter is supposed to stop frames from executing effects if the movement didn't work or vice versa.